sdk/scheduler/src/main/java/com/mesosphere/sdk/curator/CuratorStateStore.java

```diff
@@ -284,7 +284,7 @@ public void clearTask(String taskName) throws StateStoreException {
         try {
             byte[] bytes = curator.get(path);
             if (bytes.length > 0) {
-                return Optional.of(Protos.TaskInfo.parseFrom(bytes));
+                return Optional.of(CommonTaskUtils.unpackTaskInfo(Protos.TaskInfo.parseFrom(bytes)));
```

Comment:
I thought about this and went back and forth on it.  In some sense I'd like the StateStore to reflect exactly what we told Mesos, so we can debug if necessary.  So I think storing the packed information is the right thing to do.  We send packed info to Mesos, so that's what we store.